lib/matplotlib/colorbar.py

```diff
@@ -818,7 +818,7 @@ def _process_values(self, b=None):
             if self._extend_upper():
                 b = b + [b[-1] + 1]
             b = np.array(b)
-            v = np.zeros((len(b) - 1,), dtype=float)
+            v = np.zeros(len(b) - 1)
```

Comment:
Is that reasonable? This creates an `int` array which is later cast to a `float` array.

Reply:
Never mind. Sorry for the noise.